ui/src/main/java/com/hyperwallet/android/ui/view/widget/DateDialogFragment.java

```diff
@@ -0,0 +1,118 @@
+        DateDialogFragment dateDialogFragment = new DateDialogFragment();
+        Bundle bundle = new Bundle();
+        bundle.putString(ARGUMENT_DATE, date);
+        bundle.putString(ARGUMENT_FIELD_NAME, fieldName);
+        dateDialogFragment.setArguments(bundle);
+
+        return dateDialogFragment;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        try {
+            mOnSelectedDateCallback = (OnSelectedDateCallback) context;
+        } catch (ClassCastException e) {
+            throw new ClassCastException(getActivity().toString()
+                    + " must implement "
+                    + OnSelectedDateCallback.class.getCanonicalName());
+        }
+    }
+
+    public void show(@NonNull FragmentManager manager) {
+        show(manager, TAG);
+    }
+
+    @Override
+    public void onDestroyView() {
+        Dialog dialog = getDialog();
+        if (dialog != null && getRetainInstance()) {
+            dialog.setDismissMessage(null);
+        }
+        super.onDestroyView();
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle state) {
+
+        String storedDate = getArguments() != null ? getArguments().getString(ARGUMENT_DATE) : "";
+        final String fieldName = getArguments().getString(ARGUMENT_FIELD_NAME);
+        Calendar calendar;
+        try {
+            calendar = getDateUtil().convertDateFromServerFormatToCalendar(storedDate);
+        } catch (DateParseException e) {
+            calendar = Calendar.getInstance();
+        }
+
+        DatePickerDialog datePickerDialog = new DatePickerDialog(requireContext(),
+                R.style.Widget_Hyperwallet_DatePicker,
+                new DatePickerDialog.OnDateSetListener() {
+                    @Override
+                    public void onDateSet(DatePicker view, int resultYear, int resultMonth, int resultDayOfMonth) {
+                        final String selectedDate = getDateUtil()
+                                .buildDateFromDateDialogToServerFormat(resultYear, resultMonth, resultDayOfMonth);
+                        mOnSelectedDateCallback.setSelectedDateField(fieldName, selectedDate);
+                    }
+                },
+                calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH));
+
+        datePickerDialog.setButton(DialogInterface.BUTTON_NEGATIVE,
+                getString(R.string.cancel_button_label),
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        if (which == DialogInterface.BUTTON_NEGATIVE) {
+                            mOnSelectedDateCallback.setSelectedDateField(fieldName, null);
+                        }
+                    }
+                });
+
```

Comment:
The dialog will close when you touch outside of the dialog, we don't want that please add this code in line 104
`datePickerDialog.setCanceledOnTouchOutside(false);`